### PR TITLE
Instance boundary follow-up

### PR DESCRIPTION
**Pull request by Treeston**
Instance boundary follow-up to stop tools from complaining about leaks. Also, list -> vector, fixing an ambiguous constructor warning, and fixing a bug with spawn behavior on Gormok and Sindragosa (#16280).

### CODE REVIEW

**Comment by Goatform** [comment]
Wtf is this mention-bot lol ? Btw, good job @Treeston

**Comment by DDuarte** [comment]
https://github.com/facebook/mention-bot#how-does-it-work

**Comment by Goatform** [comment]
Ohh didn't knew about that. Thanks @DDuarte!

**Comment by jackpoz** [comment]
https://travis-ci.org/TrinityCore/TrinityCore/builds/102433897#L610

File: src/server/game/Maps/AreaBoundary.h [hunk 1]
```diff
@@ -40,7 +41,7 @@ class AreaBoundary
             double d_positionX, d_positionY, d_positionZ;
             DoublePosition(double x = 0, double y = 0, double z = 0, float o = 0)
                 : Position((float)x, (float)y, (float)z, o), d_positionX(x), d_positionY(y), d_positionZ(z) { }
-            DoublePosition(float x = 0, float y = 0, float z = 0, float o = 0)
+            DoublePosition(float x, float y = 0, float z = 0, float o = 0)
```
Comment (Shauren) [inline src/server/game/Maps/AreaBoundary.h]: missing f suffix on float literals

**Comment by Saben65** [comment]
Just updated the core and rebuilt and got numerous warnings. Build still succeeded. 

```
server\game\Maps\AreaBoundary.h(59): warning C4520: 'AreaBoundary::DoublePosition' : multiple default constructors specified
```

**Comment by tkrokli** [comment]
> Just updated the core and rebuilt and got numerous warnings. Build still succeeded.

... and that was with this fix included? (not 100% sure if that is what you meant)

**Comment by Saben65** [comment]
It was just pulled minutes ago, so it should include every commit.

Not sure if it specifically pertains to the updates done for the instance boundary system, but thought I would mention it since I saw AreaBoundary.h was a file that was updated.

**Comment by tkrokli** [comment]
Oh, right. That was just a copyright header update. This PR is not merged yet.

**Comment by jackpoz** [comment]
> Do You know when It'll be Availible on the Public Repo's?

Everything on github is public, what are you referring to ?

File: src/server/game/Maps/AreaBoundary.h (hunk 1 above)
Comment (jackpoz) [inline src/server/game/Maps/AreaBoundary.h]: is this some kind of trick to distinguish between double and float overloads when no arguments are passed ?

**Comment by tkrokli** [comment]
I think he meant "[...]when it'll be merged into and become part of the default source?".

File: src/server/game/Maps/AreaBoundary.h (hunk 1 above)
Comment (Treeston) [inline src/server/game/Maps/AreaBoundary.h]: Yes, it forces it into double overload when no arguments are passed.

Comment (Shauren) [inline src/server/game/Maps/AreaBoundary.h]: no, a warning fix

Comment (Shauren) [inline src/server/game/Maps/AreaBoundary.h]: tbh i think X/Y should not be defaulted for this kinda of boundary

**Comment by CDAGaming** [comment]
I'm Getting This:
Error	19	error LNK1181: cannot open input file '..\scripts\Release\scripts.lib'	E:\Trinity\Build\Build7\src\server\worldserver\LINK	worldserver

and

Error	17	error C2582: 'operator =' function is unavailable in 'BossBoundaryEntry' (E:\Trinity\Source_3.3.5_Work\src\server\scripts\Northrend\IcecrownCitadel\instance_icecrown_citadel.cpp)	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\utility	53	1	scripts

Can Someone Help Me?

**Comment by Saben65** [comment]
is your world server running?

**Comment by CDAGaming** [comment]
@Saben65 I have an Old Revision from 4 days ago so I can get On It.

@Treeston Didn't know what to do as he couldnt replicate it and all Suggestions ive gotten don't work

I'm Running VS 2013 and CMake Never gets any Errors

**Comment by CDAGaming** [comment]
Does anyone have Any Ideas BC I can't Get New Revisions Unless I can Get this issue Fixed ASAP

**Comment by CDAGaming** [comment]
I've Also Used EVERY FILE from @Treeston and Still 25 Errors in VS 2013

**Comment by Saben65** [comment]
I mean is it running when you are compiling?

**Comment by CDAGaming** [comment]
@Saben65 No I've never had it On when it's Compiling

**Comment by Saben65** [comment]
Would suggest just waiting till it gets pushed to the core, doing a fresh compile, and if the issue persists then to make a post on the forums.

**Comment by CDAGaming** [comment]
I have No-Idea when This will be Up though and I wanna get my server up for my players.

**Comment by Shauren** [comment]
ASAP my ass.

**Comment by CDAGaming** [comment]
@Shauren That's fair, Just let me know when You or Anyone is able to provide a fix to the VS error Above

**Comment by CDAGaming** [comment]
Ok, I see One of the Files has more code on it. Will apply and Recompile in an hour

**Comment by DDuarte** [comment]
@CDAGaming: You know that you can merge this PR automatically, right?

+ Diff https://github.com/TrinityCore/TrinityCore/pull/16270.diff
+ Patch https://github.com/TrinityCore/TrinityCore/pull/16270.patch
+ `git checkout -b Treeston-3.3.5-boundaryagain 3.3.5`  
`git pull https://github.com/Treeston/TrinityCore.git 3.3.5-boundaryagain`

**Comment by Treeston** [comment]
@CDAGaming You should probably not build fresh code on a production environment without keeping any backups of previous binaries. Just a thought.

We are an open source project that people work on in their free time. If you want to use it for your monetized private server, then we can't stop you from doing it, that's kind of the point of open source. However, it doesn't mean that we have to like it, and if you come in here with absolute zero idea of what you're doing and want to have your hand held through cleaning up your own mess that you got yourself into because see my first paragraph, then sorry, you can fuck right off.

&lt;/rant>

**Comment by CDAGaming** [comment]
@Treeston I'll Leave You Alone Now. I'm actually very experienced in this type of stuff but I won't bother with you anymore

**Comment by MitchesD** [comment]
@CDAGaming are you serious? At first, we are open source and learning project and TC devs work on code only in their free time, so we aren't responsible to you. Another thing, if you want to use our code on some private project, you can, but you should use master branch, this is Pull Request, it means that someone wants to share his code with others but it has to be revised, tested and whatever else. If I forget that I actually hate stupid private projects just based to earn money, there is something like decency, you CAN'T expect that someone will do what you want so please show a respect to devs and users who are trying to fix/implement something and don't bother us with your stupid questions, which you should resolve by yourself, becase for sure you have some old version of VS or whatever, we have NO compile errors and also travis passed it as well.

**Comment by CDAGaming** [comment]
@MitchesD FYI,I've Spent the Last 2 DAYS Trying Both The 3.3.5 Branch which gave me the 1st 23 Warning about multiple Default stuff, and Trying this which gave me 25 Errors of Operator=. I run VS 2013 which is Not Old and Supported by Trinity!!!

**Comment by MitchesD** [comment]
@CDAGaming FYI, VS 2013 has many versions, you didn't even say which update you are using and if you have troubles then create ticket on forum or issue tracker.

**Comment by Treeston** [comment]
Latest commit fixes #16280

**Comment by Re3os** [comment]
Sindragosa eveid after die players. confirm at https://github.com/TrinityCore/TrinityCore/commit/82c0a074b4a3dd696646e507d4b39f3b5e03cbe3

**Comment by Aokromes** [comment]
what?

**Comment by Treeston** [comment]
@Re3os That commit is not part of the PR branch. Try to reproduce the issue on df6c43bfc8a3ac6472d718c934409ce0b2045024 and, if you can, provide detailed repro steps.

**Comment by jackpoz** [comment]
is there a list of tests left that have to be done for this PR or can I wait for it to be merged before firing up my valgrind tools ?

**Comment by Treeston** [comment]
Waiting for @Shauren to see if the new `BossBoundaryData` is fine with him this time

**Comment by Re3os** [comment]
@Treeston ok ok,nice

**Comment by tkrokli** [comment]
Thanks a lot, no more warnings in the 3.3.5 compile  now. :+1:

**Pull request merged**